src/query/api/v1/handler/prometheus/remote/write.go

```diff
@@ -318,7 +329,14 @@ func (h *PromWriteHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 			numRegular        int
 			numBadRequest     int
 		)
+		logger := logging.WithContext(r.Context(), h.instrumentOpts)
 		for _, err := range errs {
+			if err.Error() == serialize.ErrTagLiteralTooLong.Error() {
+				// NB: comparing this by string value since the error might have crossed the network
+				// and thus might not be comparable using 'errors.Is()'. This might break if the error
+				// is renamed somewhere in the stack (e.g. by using 'xerrors.NewRenamedError()').
```

Comment:
Why not log the error where it originates? Then we don't have to worry about it crossing the network.

Reply:
The whole `x/serialize` package has no logging, and the `TagEncoder::Encode` method is used in multiple files. So it seemed cleaner to me to add the logging here rather than introducing logging in `TagEncoder::Encode` implementation and/or updating all the places where this method is used.

Reply:
Will adding the check in this endpoint handler cover all the places where this method is used?